File: java/common/src/main/java/com/ais/middleware/common/events/prs/AppraisalStatusUpdateFailedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published when the AppraisalReceivedSaga fails (gateway timeout, retry exhaustion, etc.).
 * Message is routed to DLQ after this event.
 *
 * Topic: prs.events.appraisal-status-update-failed
 */
public record AppraisalStatusUpdateFailedEvent(
        String correlationId,
        String appraisalId,
        String policyNumber,
        String failureReason,
        String failedStep,
        OffsetDateTime failedAt
) {
    @JsonCreator
    public AppraisalStatusUpdateFailedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("appraisalId") String appraisalId,
            @JsonProperty("policyNumber") String policyNumber,
            @JsonProperty("failureReason") String failureReason,
            @JsonProperty("failedStep") String failedStep,
            @JsonProperty("failedAt") OffsetDateTime failedAt
    ) {
        this.correlationId = correlationId;
        this.appraisalId = appraisalId;
        this.policyNumber = policyNumber;
        this.failureReason = failureReason;
        this.failedStep = failedStep;
        this.failedAt = failedAt;
    }
}

File: dotnet/Middleware.Contracts/Events/AppraisalStatusUpdateFailedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — saga failed (timeout or gateway error)
public sealed class AppraisalStatusUpdateFailedEvent
{
    public string AppraisalId { get; set; } = string.Empty;
    public string PolicyNumber { get; set; } = string.Empty;
    public string InspectionId { get; set; } = string.Empty;
    public string FailureReason { get; set; } = string.Empty;
    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset FailedAt { get; set; }
}
